### PR TITLE
Emit OS specific line endings for `prn` and `println`

### LINKED ISSUE #810: native os line endings for the print functions
Hi,

the native line endings on MS-Windows are `\r\n` but basilisp uses `\n` across all OS.

To reproduce
1. On MS-Windows, open up the REPL and invoke `println` and `prn`, line endings are `\n`: 
``` clojure
basilisp.user=> (with-out-str (println))
"\n"

basilisp.user=> (with-out-str (prn))
"\n"
```

contrast this with Clojure
``` clojure
user> (with-out-str (prn))
"\r\n"
user> (with-out-str (println))
"\r\n"
```

PR to follow.

### PR DESCRIPTION
**Pull request by ikappaki**
Hi,

can you please consider patch to emit OS specific line endings for print fns. It fixes #810.

thanks

### CODE REVIEW

File: tests/basilisp/cli_test.py [hunk 1]
```diff
@@ -185,7 +185,7 @@ def test_newline(self, run_cli):
 
     def test_simple_expression(self, run_cli):
         result = run_cli(["repl"], input="(+ 1 2)")
-        assert "basilisp.user=> 3\nbasilisp.user=> " == result.out
+        assert f"basilisp.user=> 3\nbasilisp.user=> " == result.out
```
Comment (chrisrink10) [inline tests/basilisp/cli_test.py:188]: Does this need `\n` changed to `{os.linesep}` or should the `f` string be removed?

Comment (ikappaki) [reply tests/basilisp/cli_test.py]: Sorry this was unintentional, removed.

**Review by chrisrink10 — DISMISSED** [review]
Just needs a rebase from main, but the change looks good :+1:

**Comment by chrisrink10** [comment]
Can you rebase the changelog here so we can merge this?

**Comment by ikappaki** [comment]
> Can you rebase the changelog here so we can merge this?

Done thanks

**Review by chrisrink10 — APPROVED** [review]

**Pull request merged**